src/kinesis/producer.py

```diff
@@ -46,8 +46,6 @@ def sizeof(obj, seen=None):
 
 class AsyncProducer(SubprocessLoop):
     """Async accumulator and producer based on a multiprocessing Queue"""
-    # Tell our subprocess loop that we don't want to terminate on shutdown since we want to drain our queue first
-    TERMINATE_ON_SHUTDOWN = False
```

Comment:
This is not required, as the shutdown call will correctly trigger a call to `end` which will flush the queue. With this set to true the subprocess never exits an python runs forever.